File: Math/442-binaryMethodToSolve.py


def f():
    l,r=1,2048
    while l<=r:
        m=(l+r)//2
        mm=pow(m,m)
        if mm==pow(2,2048):
            return m
        if mm>pow(2,2048):
            r=m-1
        else:
            l=m+1
            
print(f())

File: Math/442-binaryMethodToSolve.cpp
#include<iostream>
#include<cmath>
int getValues()
{
    int l=1,h=2048;
    long long target=std::pow(2,2048);
    while(l<=h)
    {
        int m=(l+h)/2;
        long long unsigned mm=std::pow(m,m);
        if(mm==target)
            return m;
        if(mm>target)
            h=m-1;
        else
            l=m+1;
    }
    return -1;
}

int main()
{
    std::cout<<getValues();
}